src/main/java/com/zepben/cimbend/network/tracing/ConnectivityResult.kt

```diff
@@ -0,0 +1,120 @@
+    }
+
+    /**
+     * @return The terminal from which the connectivity was requested.
+     */
+    fun fromTerminal(): Terminal {
+        return fromTerminal
+    }
+
+    /**
+     * @return Convenience method for getting the conducting equipment that owns the toTerminal.
+     */
+    fun to(): ConductingEquipment? {
+        return toTerminal.conductingEquipment
+    }
+
+    /**
+     * @return The terminal which is connected to the requested terminal.
+     */
+    fun toTerminal(): Terminal {
+        return toTerminal
+    }
+
+    /**
+     * @return The nominal phases that are connected in the fromTerminal.
+     */
+    fun fromNominalPhases(): List<SinglePhaseKind> {
+        return nominalPhasePaths.stream().map { obj: NominalPhasePath -> obj.from() }.collect(Collectors.toList())
+    }
+
+    /**
+     * @return The nominal phases that are connected in the toTerminal.
+     */
+    fun toNominalPhases(): List<SinglePhaseKind> {
+        return nominalPhasePaths.stream().map { obj: NominalPhasePath -> obj.to() }.collect(Collectors.toList())
+    }
+
+    /**
+     * @return The mapping of nominal phase paths between the from and to terminals.
+     */
+    fun nominalPhasePaths(): List<NominalPhasePath> {
+        return nominalPhasePaths
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as ConnectivityResult
+
+        if (fromTerminal != other.fromTerminal) return false
+        if (toTerminal != other.toTerminal) return false
+        if (nominalPhasePaths != other.nominalPhasePaths) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = fromTerminal.hashCode()
+        result = 31 * result + toTerminal.hashCode()
+        result = 31 * result + nominalPhasePaths.hashCode()
+        return result
+    }
+
+    override fun toString(): String {
+        return "ConnectivityResult{" +
+            "fromTerminal=${fromTerminal.mRID}" +
+            ", toTerminal=${toTerminal.mRID}" +
+            ", nominalPhasePaths=$nominalPhasePaths" +
+            '}'
+    }
+
+    companion object {
+        /**
+         * @param fromTerminal The terminal for which the connectivity was requested.
+         * @param toTerminal   The terminal which is connected to the requested terminal.
+         * @return The ConnectivityResult Builder to use to construct the result.
+         */
+        fun between(fromTerminal: Terminal, toTerminal: Terminal, nominalPhasePaths: Collection<NominalPhasePath>): ConnectivityResult {
+            return ConnectivityResult(fromTerminal, toTerminal, nominalPhasePaths)
+        }
+    }
+
+    init {
+        this.nominalPhasePaths = ArrayList(nominalPhasePaths)
+        this.nominalPhasePaths.sortWith(Comparator.comparing { obj: NominalPhasePath -> obj.from() }.thenComparing { obj: NominalPhasePath -> obj.to() })
```

Comment:
Could be simplified to something like (I think these are valid kotlin lib calls):
```kotlin
this.nominalPhasePaths = nominalPhasePaths.sortedWith(compareBy { it.from }.thenBy { it.to() })
```